test-reporter: Improve stack trace readability (#7327)
NomicFoundation/hardhat: .changeset/polite-mails-grab.md

---
"@nomicfoundation/hardhat-node-test-reporter": patch
---

> Fix test error logs being cut off. The default is now 5 lines (up from 3). In CI environments, it shows 100 lines for more detail.

Reviewer: ```suggestion
Fix test error `cause` chains being cut off. The default is now 5 `cause`s (up from 3). In CI environments, it's 100.
```

Reviewer: It's not lines, but `cause`s.

That means `error.cause.cause.cause...`.

Note that I already switched the `5` for `10`.
